lib/nd_list.dart

```diff
@@ -39,22 +80,132 @@ class NDList<X> {
   final List<int> _shape = [];
 
   List toIteratedList() {
-    return _list.reshape(shape);
+    // Note! Originally from tflite_flutter's ListShape extension.
```

Comment:
I'm having trouble with `toString()` which stems from `toIteratedList()`. When running:
```
import 'package:nd_list/nd_list.dart';

void main() {
  final ndList = NumNDList.zeros([2, 3, 4]);
  ndList.toString();
}
```
I get the following error:
```
Unhandled exception:
Invalid argument(s): Shape and index must have the same length
#0      getLinearIndex (package:nd_list/nd_list.dart:16:5)
#1      NDList._reshape3.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:nd_list/nd_list.dart:146:27)
#2      new _GrowableList.generate (dart:core-patch/growable_array.dart:136:28)
#3      NDList._reshape3.<anonymous closure>.<anonymous closure> (package:nd_list/nd_list.dart:143:21)
#4      new _GrowableList.generate (dart:core-patch/growable_array.dart:136:28)
#5      NDList._reshape3.<anonymous closure> (package:nd_list/nd_list.dart:141:19)
#6      new _GrowableList.generate (dart:core-patch/growable_array.dart:136:28)
#7      NDList._reshape3 (package:nd_list/nd_list.dart:139:45)
#8      NDList.toIteratedList (package:nd_list/nd_list.dart:103:18)
#9      NDList.toString (package:nd_list/nd_list.dart:208:12)
#10     main (file:///C:/Users/tavel/Desktop/Internship-Arcascope/2024_internship/nd_list/test/franco_test.dart:5:10)
#11     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#12     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```

I think it's worth adding tests for this new functionality and all the reshapes

Reply:
Ah, I expect that would be my attempt to copy what tflite_flutter does to reshape lists. The only reason we had that dependency was for their reshaping. So I probably didn't do that correctly. And literally it's just to print it nicely.

As far as our reshaping, yes we should test that. Especially the -1 functionality. Reshaping our arrays are actually really straightforward.

Reply:
This should be fixed now, give it a try.

Reply:
Works on my end!